spec/models/shared_ownership_housing_costs_form_spec.rb

```diff
@@ -0,0 +1,120 @@
+require "rails_helper"
+
+RSpec.describe SharedOwnershipHousingCostsForm do
+  let(:form) { described_class.new(attributes, Check.new) }
+
+  let(:base_attributes) do
+    {
+      mortgage: 0,
+      mortgage_frequency: nil,
+      rent: 0,
+      rent_frequency: nil,
+      housing_benefit_relevant: false,
+    }
+  end
+
+  context "when form is submitted and" do
+    context "when mortgage is positive" do
```

Comment:
@willc-work Can we include some tests to check that the session_data is actually updated with the correct values? When I run this locally it looks like possibly the mortgage isn't being saved to the session_data.